src/app/core/feedback/feedback.service.js

```diff
@@ -61,37 +61,28 @@ const create = async (reqUser, newFeedback, userSpec) => {
 	}
 };
 
-const search = async (reqUser, queryParams, query) => {
+const search = (reqUser, queryParams, search, query) => {
 	query = query || {};
 	const page = util.getPage(queryParams);
 	const limit = util.getLimit(queryParams, 100);
-	const sortArr = util.getSort(queryParams);
-	const offset = page * limit;
+	const sort = util.getSortObj(queryParams);
 
 	// Query for feedback
-	const feedback = await Feedback.textSearch(
-		query,
-		null,
-		limit,
-		offset,
-		sortArr,
-		true,
-		{
+	return Feedback.find(query)
+		.textSearch(search)
+		.sort(sort)
+		.populate({
 			path: 'creator',
 			select: ['username', 'organization', 'name', 'email']
-		}
-	);
-
-	return util.getPagingResults(limit, page, feedback.count, feedback.results);
+		})
+		.paginate(limit, page);
```

Comment:
This shows the advantage of this change.  Instead of having to use a single function that takes every possible parameter we may want to pass in to build our query, we can now use the mongoose query builder API.